### DISCUSSION TITLE
Create with condition on nested relation

### DISCUSSION THREAD

**Discussion by verekia** [prisma/prisma#7155] [Q&A] [answered]
```prisma
model User {
  id     String @id
  notes  Note[]
}

model Note {
  id           String        @id
  userId       String
  user         User          @relation(fields: [userId], references: [id])
  attachments  Attachments[]
}

model Attachment {
  id      String @id
  noteId  String
  note    Note   @relation(fields: [noteId], references: [id])
}
```
Here, a `User` has `notes`, and those notes have `attachments`.
I try to avoid storing `userId` on `Attachement`, because since an attachment knows its `Note`, by extension it knows its `User`.

I am trying to make a query to **create an Attachment only if the provided userId matches the userId of the related Note**. Basically, I want to check that the Note we're attaching an Attachment to is the user's, and not someone else's.

With updates or deletes, I can do something like `where: { id, note: { id: noteId, user: { id: userId } } }`, but that's not possible on a create.

My currently workaround is to do 2 queries, first, retrieve the `Note`, then if the `note.userId` matches the one of the user, allow the `create`.

Is there any way to include this condition and do this in 1 query? And in general adding conditions to creates?

**Comment by ryands17** [comment, marked as answer]
@verekia 👋 

Unfortunately 2 queries would be required in this case. I would suggest following this request: https://github.com/prisma/prisma/issues/3143 as the update would be then possible in a single request via something like this:

```ts
await prisma.note.updateMany({ 
  where: { id: '1', userId: '1' }, 
  data: {
    // create attachment here
 }
})
```

**Reply by verekia** [reply]
Thank you very much!
